compiler/noirc_frontend/src/hir/def_map/namespace.rs

```diff
@@ -3,13 +3,13 @@ use super::{item_scope::Visibility, ModuleDefId};
 // This works exactly the same as in r-a, just simplified
 #[derive(Debug, PartialEq, Eq, Copy, Clone)]
 pub struct PerNs {
-    pub types: Option<(ModuleDefId, Visibility)>,
-    pub values: Option<(ModuleDefId, Visibility)>,
+    pub types: Option<(ModuleDefId, Visibility, bool)>,
+    pub values: Option<(ModuleDefId, Visibility, bool)>,
```

Comment:
To avoid the error "x is already defined" when importing a name that has already been imported from Prelude

Reply:
Is this to stop these errors because we import the prelude into the prelude or because we want to allow users to define their own `Vec`/`Option` types or `print` and `println` functions?

Reply:
Re-import types from prelude. That is, there is existing code that manually imports Vec from std, but if we add prelude, there will be an error due to two Vecs being imported. (and a strange error arises because the code does not exist in the code).
We can't import prelude into prelude because we have a check that it's not a standard library. We don't add prelude to std.

Reply:
Can't we determine this by looking at the path of the import?

Reply:
It seems that at this stage we don't have Path, there is ident (for example, Option) and PerNs. The possibility to know that this is use std::prelude::*, it seems not

Reply:
Yeah, this is a different struct to what I was looking at. `ImportDirective`  can do this.

Reply:
Can we close this discussion?